### PR TITLE
Mv no-ecx and no-ecdsa into no-ec

### PR DESCRIPTION
**Pull request by richsalz**
I am not sure if we should do this, unless we also agree to remove no-ec/ec as a config option.

Thoughts?

See also https://github.com/openssl/openssl/issues/19127#issuecomment-1238211958

## Checklist
Thank you for your contribution. Please answer the following:

- [X] I acknowledge that I am authorized to submit this code under
the terms of the [Apache License](https://www.apache.org/licenses/LICENSE-2.0)
- [ ] All new public APIs are documented
- [ ] All new public functions have tests
- [ ] All new public functions have fuzzing tests

### CODE REVIEW

**Comment by botovq** [comment]
It makes very little sense to me to disable either EC or ECX in a modern TLS/QUIC library. Some of the config options were useful to disable old decrepit stuff (much of which was moved to the legacy provider). What is considered state of the art (non-PQ) crypto should not need disabling in the foreseeable future.

**Comment by richsalz** [comment]
I agree.  Let's see if anyhone disagrees.  I propose we merge this, and then remove no-ec and no-dh and no-eddsa all in one go.

**Comment by botovq** [comment]
Just to be explicit: If the decision is to remove them, then yes, this makes sense as a first step and I will approve the PR. Otherwise I think it is better to leave them separate.

**Comment by botovq** [comment]
Apparently there are conflicts here, once they are resolved, I'll approve.

**Comment by richsalz** [comment]
rebased and fixed the conflict in Configure.

**Comment by richsalz** [comment]
Added the two lines of `OPENSSL_NO_ECDSA` to this PR

**Comment by richsalz** [comment]
Rebased, fixed merge conflcits, pushed.

File: apps/list.c [hunk 1]
```diff
@@ -1343,8 +1343,6 @@ static void list_disabled(void)
 #endif
 #ifdef OPENSSL_NO_EC
     BIO_puts(bio_out, "EC\n");
-#endif
-#ifdef OPENSSL_NO_ECX
     BIO_puts(bio_out, "ECX\n");
```
Comment (botovq) [inline apps/list.c:1346]: ECX is no longer a thing, so this should be removed

File: test/recipes/70-test_tls13cookie.t [hunk 2]
```diff
@@ -44,11 +44,11 @@ my $testtype;
 #Test 1: Inserting a cookie into an HRR should see it echoed in the ClientHello
 $testtype = COOKIE_ONLY;
 $proxy->filter(\&cookie_filter);
-$proxy->serverflags("-curves X25519") if !disabled("ecx");
+$proxy->serverflags("-curves X25519") if !disabled("ec");
 $proxy->start() or plan skip_all => "Unable to start up Proxy for tests";
 plan tests => 2;
 SKIP: {
-    skip "ECX disabled", 1, if (disabled("ecx"));
+    skip "ECX disabled", 1, if (disabled("ec"));
```
Comment (botovq) [inline test/recipes/70-test_tls13cookie.t:51]:
```suggestion
    skip "EC disabled", 1, if (disabled("ec"));
```

File: test/recipes/20-test_pkeyutl.t [hunk 3]
```diff
@@ -55,7 +55,7 @@ SKIP: {
 
 SKIP: {
     skip "Skipping tests that require ECX", 4
```
Comment (botovq) [inline test/recipes/20-test_pkeyutl.t:57]:
```suggestion
    skip "Skipping tests that require EC", 4
```

File: test/evp_extra_test.c [hunk 4]
```diff
@@ -4990,7 +4988,7 @@ static int test_signatures_with_engine(int tst)
     if (tst <= 1)
         return 1;
 #  endif
-#  ifdef OPENSSL_NO_ECX
+#  ifdef OPENSSL_NO_EC
     /* Skip ECX tests in a no-ecx build */
```
Comment (botovq) [inline test/evp_extra_test.c:4992]:
```suggestion
    /* Skip ECX tests in a no-ec build */
```

**Review by botovq — CHANGES_REQUESTED** [review]
There are a bunch of `providers/implementations/*/build.info` files that still do
```
IF[{- !$disabled{ecx} -}]
```

File: apps/list.c (hunk 1 above)
Comment (richsalz) [reply apps/list.c:1346]: will fix.

**Comment by richsalz** [comment]
Rebased, resolved conflicts, pushed.

File: providers/common/der/build.info [hunk 5]
```diff
@@ -34,10 +34,9 @@ IF[{- !$disabled{ec} -}]
 
   DEPEND[${DER_EC_AUX/.c/.o}]=$DER_EC_H $DER_DIGESTS_H
   DEPEND[${DER_EC_GEN/.c/.o}]=$DER_EC_H
-ENDIF
+  GENERATE[$DER_EC_H]=$INCDIR/der_ec.h.in
+  DEPEND[$DER_EC_H]=oids_to_c.pm EC.asn1
```
Comment (botovq) [inline providers/common/der/build.info:38]: These two lines should be removed - looks like an incorrectly resolved conflict. Other than that this looks good now.

Comment (richsalz) [reply providers/common/der/build.info]: And I only did PR #160 today :). Sigh.  fixup commit pushed. thanks.

**Review by botovq — APPROVED** [review]

**Pull request merged**